### PR TITLE
changes a couple of things to get the CPython binding working.

### PR DESCRIPTION
**Pull request by kdschlosser**
These are pretty minor changes. One is removing the declaration for the `_lv_font_clean_up_fmt_txt` function. The function wasn't defined which caused linking problems.

I also moved a couple of structures that were defined in source files and put them into the header files. If this is an issue let me know, I will see if I can manage to work around them not being exposed. I am not sure if ctypes will play nice with me if it is unable to determine the size of the structures. Because ctypes is disconnected so to speak from the structures in C code The only way for me to show their existence is if they can be seen in the header files. That includes the fields in the structures. This is if the structure is used aas a type for aa field in a public structure or as a type in a public function.

They are small changes, I hope it's not a big deal to make them. I managed to code around the rest of the issues I was having before.

The code generation script is too specific to LVGL for my liking, so it will need maintenance due to it having specific parts of it coded specifically to handle LVGL, if those things change it could break. Not a huge deal.

### CODE REVIEW

File: src/misc/lv_event.h [hunk 1]
```diff
@@ -25,9 +25,15 @@ extern "C" {
 /**********************
  *      TYPEDEFS
  **********************/
+typedef void (*lv_event_cb_t)(struct _lv_event_t * e);
```
Comment (kisvegabor) [inline src/misc/lv_event.h:30]: To solve the error:
```suggestion
struct _lv_event_t;

typedef void (*lv_event_cb_t)(struct _lv_event_t * e);
```

File: src/misc/lv_anim_timeline.h [hunk 2]
```diff
@@ -23,9 +23,18 @@ extern "C" {
  *      TYPEDEFS
  **********************/
 
-struct _lv_anim_timeline_t;
-
-typedef struct _lv_anim_timeline_t lv_anim_timeline_t;
+ /*Data of anim_timeline_dsc*/
```
Comment (kisvegabor) [inline src/misc/lv_anim_timeline.h]:
```suggestion
/*Data of anim_timeline_dsc*/
```

**Comment by kdschlosser** [comment]
OK so I am not sure what the "resolved" means. LOL

**Comment by kisvegabor** [comment]
Thanks, all looks good. I've fixed a simple error, that's why you see it resolved. :slightly_smiling_face:

**Comment by kdschlosser** [comment]
Funny tho it works without declaring the struct ahead of time like that.

Is that the reason why the compiler warning existed?

**Comment by kisvegabor** [comment]
It's really strange that it worked for you. I thought it should fail to compile. Sometimes miracles happen :slightly_smiling_face:

**Comment by kdschlosser** [comment]
Depends on of you have compiler warning set to become errors I guess. with MSVC it might be a warning. who knows. It did compile tho. MSVC is so far off from C** compliance it's crazy. The hoops you have to jump through just to get functions exported where gcc exports all function right out of the box.

**Pull request merged**